selfdrive/ui/qt/widgets/controls.h

```diff
@@ -127,3 +126,28 @@ class ParamControl : public ToggleControl {
 private:
   Params params;
 };
+
+class ListWidget : public QWidget {
```

Comment:
why not make this a layout?

Reply:
I think do painting  is better, performance is better too.It can save a lot of styled horizontal_line widgets. The layouting is faster, and the spacing between controls can be controlled more precisely.

Reply:
we should get the better perf if make a horizontal line widget drawn with paintEvent. the reason I'd like this to be a layout is because this looks like pseudo-Qt code with a weird widget class that has the API of a layout, but it really seems like there's no good reason for that. I think we can get what we want, and it can still feel like familiar Qt code

Reply:
it's more looks like a QListWidget if we remove the function addLayout. what do you think? addLayout is only used in one place

Reply:
An extra line widget is difficult to control the spacing between controls. a bool variable also needs to be added to let the caller tell us which one is the last.we can think this widget as a  styled qlistwidget with bottom line between items